src/hotspot/cpu/riscv/assembler_riscv.hpp

```diff
@@ -2095,10 +2095,19 @@ enum Nf {
 //    but most of time we have no need to explicitly use these instructions.
 // 3. 'CompressibleRegion' is introduced to hint instructions in this Region's RTTI range
 //    are qualified to be compressed with their 2-byte versions.
+//    'IncompressibleRegion' hints instructions to remain their 4-byte forms.
 //    An example:
 //
 //      CompressibleRegion cr(_masm);
-//      __ andr(...);      // this instruction could change to c.and if able to
+//      __ andr(...);       // this instruction could change to 'c.and' if able to
+//      {
+//         IncompressibleRegion ir(_masm);
+//         __ andr(...);    // this instruction should not be compressed
+//         {
+//            CompressibleRegion cr(_masm);
+//            __ andr(...); // this instruction could change to 'c.and' if able to
+//         }
+//      }
 //
 // 4. Using -XX:PrintAssemblyOptions=no-aliases could distinguish RVC instructions from
 //    normal ones.
```

Comment:
One more suggestion for this code comment:

// ========================================
// RISC-V Compressed Instructions Extension
// ========================================
// Note:
// 1. Assembler functions encoding 16-bit compressed instructions always begin with a 'c_'
//    prefix, such as 'c_add'. Correspondingly, assembler functions encoding normal 32-bit
//    instructions with begin with a '_' prefix, such as "_add". Most of time users have no
//    need to explicitly emit these compressed instructions. Instead, they still use unified
//    wrappers such as 'add' which do the compressing work through 'c_add' depending on the
//    the operands of the instruction and availability of the RVC hardware extension.
//
// 2. 'CompressibleRegion' and 'IncompressibleRegion' are introduced to mark assembler scopes
//     within which instructions are qualified or unqualified to be compressed into their 16-bit
//     versions.
//
// 3. When the RVC hardware extension is available, normal 32-bit instructions within scope
//    with a 'CompressibleRegion' mark will be compressed to their 16-bit versions when the
//    operands are valid.
//
//    An example:
//
//      CompressibleRegion cr(_masm);
//      __ add(...);       // this instruction will be compressed into 'c.and' when possible
//      {
//         IncompressibleRegion ir(_masm);
//         __ add(...);    // this instruction will not be compressed
//         {
//            CompressibleRegion cr(_masm);
//            __ add(...); // this instruction will be compressed into 'c.and' when possible
//         }
//      }
//
// 4. When printing JIT assembly code, using -XX:PrintAssemblyOptions=no-aliases could help
//    distinguish compressed 16-bit instructions from normal 32-bit ones.

Reply:
Thanks for the polish and it looks more concise than before. The other one, the code style of nop is fixed as well.